mlflow/openai/__init__.py

```diff
@@ -576,22 +576,66 @@ def _has_content_and_role(d):
     return "content" in d and "role" in d
 
 
-class _FormattableMessage:
-    def __init__(self, message):
-        self.content = message.get("content")
-        self.role = message.get("role")
-        self.variables = _parse_format_fields(self.content)
+class _ContentFormatter:
+    def __init__(self, type, template=None):
+        if type == "completions":
+            if not template:
+                template = "{prompt}"
+            if not isinstance(template, str):
+                raise mlflow.MlflowException.invalid_parameter_value(
+                    f"Type {type(template)} is not a valid type for template of task {type}."
+                )
+
+            self.template = template
+            self.format_fn = self.format_prompt
+            self.variables = sorted(_parse_format_fields(self.template))
+        elif type == "chat.completions":
+            if not template:
+                template = [{"role": "user", "content": "{content}"}]
+            if not all(isinstance(message, Dict) for message in template):
+                raise mlflow.MlflowException.invalid_parameter_value(
```

Comment:
Do you feel that it's important to check the value types as well to preempt an obscure exception if, say, a float is passed?

Reply:
I can replace that check for `if not all(_has_content_and_role(message) for message in template):` which checks for the keys we need. What do you think?

Reply:
I think the check that's in place is great and it perfectly captures the validation of the container :) 
I was just wondering if you thought that a type validation would be needed for those values. What happens if we pass a list of ints for `content`? Is there a meaningful exception thrown? Does it optimistically cast and return something interesting? Or is it a confusing mess for a user?

Reply:
Passing `content="1 2 3 4 5 6"` should be a valid thing. But agree we can validate that the associated types are string!